projects/cm_mcu/I2CCommunication.c

```diff
@@ -107,17 +108,20 @@ int apollo_i2c_ctl_r(uint8_t device, uint8_t address, uint8_t nbytes, uint8_t da
   return retval;
 }
 
-int apollo_i2c_ctl_reg_r(uint8_t device, uint8_t address, uint8_t reg_address, uint8_t nbytes, uint8_t data[MAX_BYTES])
+int apollo_i2c_ctl_reg_r(uint8_t device, uint8_t address, uint8_t nbytes_addr, uint16_t packed_reg_address, uint8_t nbytes, uint32_t packed_data)
 {
   tSMBus* smbus = pSMBus[device];
   tSMBusStatus* p_status = eStatus[device];
 
   configASSERT(smbus != NULL);
-
-  memset(data, 0, nbytes * sizeof(data[0]));
-  if (nbytes > MAX_BYTES)
-    nbytes = MAX_BYTES;
-
+  uint8_t reg_address[MAX_BYTES_ADDR];
+    for (int i = 0; i < MAX_BYTES_ADDR; ++i) {
```

Comment:
please fix formatting here. use spaces, not tabs

Reply:
fixed it.